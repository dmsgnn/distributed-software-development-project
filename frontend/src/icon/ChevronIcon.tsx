import { Icon, IconProps } from '@chakra-ui/react';
import React, { FC } from 'react';

export const ChevronIcon: FC<IconProps> = (props) => {
	return (
		<Icon color="primary.500" viewBox="0 0 24 24" {...props}>
			<path
				d="M16 4.4c.3.3.4.6.4 1s-.1.7-.4 1L10.4 12l5.6 5.6c.3.3.4.6.4 1s-.1.7-.4 1c-.3.3-.6.4-1 .4s-.7-.1-1-.4L7.4 13c-.1-.1-.3-.3-.3-.5a1.6 1.6 0 0 1 0-1c0-.2.2-.4.3-.5L14 4.4c.3-.3.6-.4 1-.4s.7.1 1 .4Z"
				fill="currentColor"
			/>
		</Icon>
	);
};
