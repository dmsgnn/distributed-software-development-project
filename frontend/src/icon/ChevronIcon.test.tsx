import { axe } from 'jest-axe';

import { act, render } from 'test-utils';
import { ChevronIcon } from './ChevronIcon';

describe('ChevronIcon', () => {
	it('should matches snapshot', () => {
		const { asFragment } = render(<ChevronIcon />);

		expect(asFragment()).toMatchSnapshot();
	});

	it('should be accessible', async () => {
		const { container } = render(<ChevronIcon />);

		let results: Awaited<ReturnType<typeof axe>>;
		await act(async () => {
			results = await axe(container);
		});

		expect(results).toHaveNoViolations();
	});
});
