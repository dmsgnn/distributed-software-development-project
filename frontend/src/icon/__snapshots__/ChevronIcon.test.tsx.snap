// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChevronIcon should matches snapshot 1`] = `
<DocumentFragment>
  <svg
    class="chakra-icon css-q5208i"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M16 4.4c.3.3.4.6.4 1s-.1.7-.4 1L10.4 12l5.6 5.6c.3.3.4.6.4 1s-.1.7-.4 1c-.3.3-.6.4-1 .4s-.7-.1-1-.4L7.4 13c-.1-.1-.3-.3-.3-.5a1.6 1.6 0 0 1 0-1c0-.2.2-.4.3-.5L14 4.4c.3-.3.6-.4 1-.4s.7.1 1 .4Z"
      fill="currentColor"
    />
  </svg>
</DocumentFragment>
`;
